src/editors/containers/ProblemEditor/components/EditProblemView/index.jsx

```diff
@@ -25,16 +27,16 @@ export const EditProblemView = ({
   return (
     <EditorContainer getContent={getContent}>
       <div className="editProblemView d-flex flex-row flex-nowrap justify-content-end">
-        <span className="flex-grow-1">
-          {isAdvancedProblemType ? (
+        {isAdvancedProblemType ? (
+          <Container size="lg" className="mt-4.5">
             <RawEditor editorRef={editorRef} lang="xml" content={problemState.rawOLX} />
-          ) : (
-            <>
-              <QuestionWidget />
-              <AnswerWidget problemType={problemType} />
-            </>
-          )}
-        </span>
+          </Container>
+        ) : (
+          <span className="flex-grow-1">
+            <QuestionWidget />
+            <AnswerWidget problemType={problemType} />
+          </span>
+        )}
         <span className="editProblemView-settingsColumn">
```

Comment:
I had a look at the actual reason the settings widget (or span container around it) gets too small. So it already has a (correct) flex-basis of 320px. Of course, that may shrink when the question editor expands in an unexpected way. So adding a `flex-shrink: 0` and maybe also `width: 320px` seems to keep it from shrinking so it stays correct. Additionally, there is some margin on the left side of the Settingswidget inside the flex element that takes up some of that 320px - we need to remove that and maybe add it in another place so we keep the correct margin from figma.
Where to make those changes: the class `editProblemView-settingsColumn` should be in the corresponding .scss file, which probably needs to be adjusted.